partner_communication/migrations/10.0.1.2.0/post-migration.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2019 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+from openupgradelib import openupgrade
+
+
+# Warning: This migration can take a long time!
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    if not version:
+        return
+    cr = env.cr
+
+    cr.execute("ALTER TABLE partner_communication_attachment "
+               "ADD COLUMN IF NOT EXISTS report_id INTEGER")
+
+    cr.execute("SELECT id FROM partner_communication_attachment")
+    communication_ids = cr.dictfetchall()
+
+    for communication_id in communication_ids:
```

Comment:
You don't need all the `cr` and query thing. Here you want to update the communication attachments so I would just do this :
```
attachments = env['partner.communication.attachment'].search([
    ('report_name', '!=', False),
    ('communication_id', '!=', False)
])
for attachment in attachments:
    attachment.report_id = \
            env['report']._get_report_from_name(attachment.report_name)
```